PubNubUnity/Assets/PubNub/Helpers/TokenHelpers.cs

```diff
@@ -0,0 +1,411 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using PeterO.Cbor;
+using System.Text;
+
+
+namespace PubNubAPI
+{
+    public enum PNGrantBitMask
+    {
+        PNRead = 1,
+        PNWrite = 2,
+        PNManage = 4,
+        PNDelete = 8,
+        PNCreate =16,
+        PNGet = 32,
+        PNUpdate = 64,
+        PNJoin = 128
+    }
+
+    public enum PNGrantType
+    {
+        PNReadEnabled = 0,
+        PNWriteEnabled,
+        PNManageEnabled,
+        PNDeleteEnabled,
+        PNCreateEnabled,
+        PNGetEnabled,
+        PNUpdateEnabled,
+        PNJoinEnabled
+    }
+
+    public enum PNResourceType{
+        PNChannels = 0,
+        PNGroups,
+        PNUUIDMetadata,
+        PNChannelMetadata,
+        PNUUIDs
+    }
+
+    public class TokenAuthValues {
+        public bool Read {get; set;}
+        public bool Write {get; set;}
+        public bool Manage {get; set;}
+        public bool Delete {get; set;}
+        public bool Create {get; set;}
+        public bool Get {get; set;}
+        public bool Update {get; set;}
+        public bool Join {get; set;}
+
+    }
+
+    public class TokenPatterns{
```

Comment:
`public class TokenPatterns` and `public class TokenResources` have same properties/fields.
Is it possible to have one class with meaning name and re-use the same class instance for both?

Reply:
Specs has two different classes

Reply:
my thought process is like
public class BaseResourceTypes { }
public class TokenPatterns : BaseResourceTypes {}
public class TokenResources : BaseResourceTypes{}
Any way, I will the leave the implementation detail to you.

Reply:
done